File: 75.cpp
struct Solution {
    void sortColors(vector<int>& nums) {
        // https://en.wikipedia.org/wiki/Dutch_national_flag_problem
        int i = 0, j = 0, k = nums.size() - 1;
        while (j <= k){
            switch (nums[j]){
                case 0:
                    swap(nums[i++], nums[j++]);
                    break;
                case 2:
                    swap(nums[j], nums[k--]);
                    break;
                case 1:
                    ++j;
                    break;
                default:
                    break;
            }
        }
    }
};

File: sol/75/75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # https://en.wikipedia.org/wiki/Dutch_national_flag_problem
        i, j = 0, 0
        k = len(nums) - 1
        while j <= k:
            if nums[j] == 0:
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j += 1
            elif nums[j] == 2:
                nums[j], nums[k] = nums[k], nums[j]
                k -= 1
            else:
                j += 1
